### PR TITLE
RavenDB-12877: Implement Rvn32BitsMemoryMapPager for posix

### PR DESCRIPTION
**Pull request by aviviadi**

### CODE REVIEW

File: src/Voron/Impl/Paging/Rvn32BitsMemoryMapPager.cs [hunk 1]
```diff
@@ -3,83 +3,87 @@
...
-            _isSyncDirAllowed = Syscall.CheckSyncDirectoryAllowed(FileName.FullPath);
-
-            PosixHelper.EnsurePathExists(FileName.FullPath);
 
-            Debug.Assert(RuntimeInformation.IsOSPlatform(OSPlatform.OSX) == false); // O_LARGEFILE not exists in mac and supported by default (however we do not run on mac 32bit..)
-
-            _fd = Syscall.open(file.FullPath, OpenFlags.O_RDWR | PerPlatformValues.OpenFlags.O_CREAT | PerPlatformValues.OpenFlags.O_LARGEFILE,
-                              FilePermissions.S_IWUSR | FilePermissions.S_IRUSR);
-            if (_fd == -1)
+//            _totalAllocationSize = GetFileSize();
```
Comment (ayende) [inline src/Voron/Impl/Paging/Rvn32BitsMemoryMapPager.cs]: Can we get rid of the commented code?

File: src/Voron/Impl/Paging/Rvn32BitsMemoryMapPager.cs [hunk 2]
```diff
@@ -247,29 +268,39 @@ private LoadedPage MapPages(TransactionState state, long startPage, long size)
...
+                        var diskSpaceResult = DiskSpaceChecker.GetDiskSpaceInfo(FileName.FullPath);
+                        throw new DiskFullException(FileName.FullPath, size, diskSpaceResult?.TotalFreeSpace.GetValue(SizeUnit.Bytes), dfEx.Message);
+                    }
                 }
 
-                NativeMemory.RegisterFileMapping(FileName.FullPath, startingBaseAddressPtr, size, GetAllocatedInBytes);
+
+                var startingBaseAddressIntPtr = new IntPtr(startingBaseAddressPtr);
+                NativeMemory.RegisterFileMapping(FileName.FullPath, startingBaseAddressIntPtr, size, GetAllocatedInBytes);
+                if (File.Exists("/tmp/debug.tmp"))
```
Comment (ayende) [inline src/Voron/Impl/Paging/Rvn32BitsMemoryMapPager.cs]: Need to remove this

File: src/Voron/Impl/Paging/Rvn32BitsMemoryMapPager.cs [hunk 3]
```diff
@@ -510,5 +540,15 @@ protected override bool CanPrefetchQuery()
             // explicitly disable prefetch
             return false;
         }
+
+        // public override void ReleaseAllocationInfo(byte* baseAddress, long size) - use base class method (32 bits is not (supporting) unmapping here
```
Comment (ayende) [inline src/Voron/Impl/Paging/Rvn32BitsMemoryMapPager.cs:527]: commented code

File: src/Raven.Pal/inc/rvn.h [hunk 4]
```diff
@@ -107,7 +107,7 @@ EXPORT int32_t
 rvn_remap(void *base_address, void **new_address, void *handle, int64_t size, int32_t flags, int64_t offset, int32_t *detailed_error_code);
 
 EXPORT int32_t
-rvn_mmap_file(int64_t sz, int64_t flags, void *handle, int64_t offset, void **addr, int32_t *detailed_error_code);
+rvn_mmap_file(int64_t sz, int32_t flags, void *handle, int64_t offset, void **addr, int32_t *detailed_error_code);
```
Comment (ayende) [inline src/Raven.Pal/inc/rvn.h:110]: Don't we need to change this on the C# as well?

File: src/Raven.Pal/src/posix/mapping.c [hunk 5]
```diff
@@ -290,22 +290,25 @@ rvn_remap(void *base_address, void **new_address, void *handle, int64_t size, in
     return rc;
 }
 
+
+/* ADIADI:: The mfh is illegal (probably passing void ** instead of void*? why fail only on the pi itself? is happening from remap?) */
```
Comment (ayende) [inline src/Raven.Pal/src/posix/mapping.c]: I don't understand this comment.

File: src/Voron/Impl/Paging/Rvn32BitsMemoryMapPager.cs (hunk 1 above)
Comment (aviviadi) [reply src/Voron/Impl/Paging/Rvn32BitsMemoryMapPager.cs]: Still in "draft" mode

File: src/Voron/Impl/Paging/Rvn32BitsMemoryMapPager.cs (hunk 2 above)
Comment (aviviadi) [reply src/Voron/Impl/Paging/Rvn32BitsMemoryMapPager.cs]: Still in "draft" mode

File: src/Voron/Impl/Paging/Rvn32BitsMemoryMapPager.cs (hunk 3 above)
Comment (aviviadi) [reply src/Voron/Impl/Paging/Rvn32BitsMemoryMapPager.cs:527]: Still in "draft" mode

File: src/Raven.Pal/inc/rvn.h (hunk 4 above)
Comment (aviviadi) [reply src/Raven.Pal/inc/rvn.h]: C# default enum is "int".. it is 32 bits integer by default.

Comment (aviviadi) [reply src/Raven.Pal/inc/rvn.h:110]: C# default enumeration is integer (C Sharp's "int") so it is 32 bits by default

**Comment by aviviadi** [comment]
Closing for "second" round (current changes will be used by @myarichuk)
@myarichuk - please bookmark this PR.

**Pull request closed without merging**